app/com/arpnetworking/metrics/portal/scheduling/TwoWayJobRefSerializer.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.metrics.portal.scheduling;
+
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.google.common.collect.ImmutableMap;
+import models.internal.impl.DefaultOrganization;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+
+/**
+ * A {@link JobRefSerializer} that allows construction to and from entity IDs.
+ *
+ * Round-trip is only guaranteed for refs whose repository types are whitelisted by this serializer
+ * (e.g. allowed by the constructor arguments).
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public class TwoWayJobRefSerializer implements JobRefSerializer {
```

Comment:
I would be fine with something like `JobRefSerializer` or `JobRefConverter` if implementing either `Converter` or both `Serializer` and `Deserializer` (see comment above). Alternatively, you could have separate `JobSerializer` and `JobRefDeserializer` implementations.